### PR TITLE
Refresh social graph periodically

### PR DESCRIPTION
**Pull request by yonatanm**
- each 6 hours - process checks which graph was not updated in the last 6 hours - and refetch it
- while re-fetching- new connections will be created, and existing connection might be de-activated if they are no longer part of the 'friends' list
- test will be added in the next PullRequest

### CODE REVIEW

**Comment by yonatanm** [comment]
I see now that this pull request contains 3 file deletion
SocialUserCreateConnectionsTest.scala, SocialUserInfoTest.scala and logger.xml

I wasn't not intended to do so. if no one else deleted it, let me know and I will pull them back.

**Comment by eishay** [comment]
yes, we still have them in the project. can you manually add them back?

File: shoebox/app/com/keepit/common/social/SocialGraphRefresherPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import securesocial.core.{SocialUser, UserId, AuthenticationMethod, OAuth2Info}
+
+private case class RefreshUserInfo(socialUserInfo: SocialUserInfo)
+private case object RefreshAll
+
+private[social] class SocialGraphRefresherActor(socialGraphPlugin : SocialGraphPlugin) extends Actor with Logging {
+  def receive() = {
+    case RefreshAll => {
+      log.info("going to check which SocilaUserInfo Was not fetched Lately")
+      val needToBeRefreshed = CX.withConnection { implicit conn => SocialUserInfo.getNeedtoBeRefreshed }
```
Comment (eishay) [inline shoebox/app/com/keepit/common/social/SocialGraphRefresherPlugin.scala]: getNeedtoBeRefreshed -> getNeedToBeRefreshed

File: shoebox/app/com/keepit/common/social/SocialGraphRefresherPlugin.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  implicit val actorTimeout = Timeout(5 seconds)
+  
+  private val actor = system.actorOf(Props { new SocialGraphRefresherActor(socialGraphPlugin) })
+  
+  // plugin lifecycle methods
+  private var _cancellables: Seq[Cancellable] = Nil
+  override def enabled: Boolean = true
+  override def onStart(): Unit = {
+    _cancellables = Seq(
+      system.scheduler.schedule(0 seconds, 2 hours, actor, RefreshAll)
```
Comment (eishay) [inline shoebox/app/com/keepit/common/social/SocialGraphRefresherPlugin.scala]: lets do it every five minutes. the query shouldn't be too hard on the db and over time we'll better spread the requests to facebook along the day

File: shoebox/app/com/keepit/common/social/SocialUserCreateConnections.scala [hunk 3]
```diff
@@ -9,26 +9,74 @@ import play.api.libs.json.JsArray
...
 
-
 class SocialUserCreateConnections() extends Logging {
 
-  def createConnections(socialUserInfo: SocialUserInfo, parentJsons: Seq[JsValue]): Seq[SocialConnection] =
-    parentJsons map {json =>
-      createConnectionsFromJson(socialUserInfo, json)
-    } flatten
+  def createConnections(socialUserInfo: SocialUserInfo, parentJson: Seq[JsValue]): Seq[SocialConnection] = {
+	disableConnectionsNotInJson(socialUserInfo, parentJson)
```
Comment (eishay) [inline shoebox/app/com/keepit/common/social/SocialUserCreateConnections.scala]: indentation ?

File: shoebox/app/com/keepit/common/social/SocialUserCreateConnections.scala [hunk 4]
```diff
@@ -9,26 +9,74 @@ import play.api.libs.json.JsArray
...
       }
     }
+  }
+  
+  def disableConnectionsNotInJson(socialUserInfo: SocialUserInfo, parentJson: Seq[JsValue]): Seq[SocialConnection] = {
+    log.info("looking for connections to disable for user %s".format(socialUserInfo.fullName))
+    CX.withConnection { implicit conn =>
+    {
+	  val socialUserInfoForAllFriendsIds = parentJson flatMap extractFriends map extractSocialId  
+	  val existingSocialUserInfoIds = SocialConnection.getUserConnections(socialUserInfo.userId.get).toSeq map {sui => sui.socialId}
```
Comment (eishay) [inline shoebox/app/com/keepit/common/social/SocialUserCreateConnections.scala]: indentation?

**Pull request merged**